roachtest: add tpcc OR roachtest

msbutler opened the pull request:
This patch adds a 400 GB and 8.5 TB Online Restore TPCC roachtest.

Epic: none

Release note: none

msbutler committed:
roachtest: add tpcc OR roachtests

This patch adds a 400 GB and a 8.5 TB (skipped) Online Restore TPCC roachtest.

Epic: none

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/123942)
<!-- Reviewable:end -->

msbutler committed:
roachtest: better exercise prefix synthesis in OR scale tests

This patch adds a few tables to the roachtest cluster before OR begins to
better exercise prefix synthesis. This patch also adds a check to ensure all
data was downloaded at the end of the download job.

Epic: none

Release note: none

msbutler committed:
roachtest: add more OR workarounds

This patch disables the consistency queue and enables
kv.range_merge.skip_external_bytes.enabled, which skips adding replicas to the
merge queue if they contain external bytes.

Epic: none

Release note: none

kev-cao reviewed (approved):
LGTM!

msbutler commented:
tftr!

bors r=kev-cao
